plugins/jobs/src/js/pages/JobDetailPage.js

```diff
@@ -0,0 +1,193 @@
+import classNames from "classnames";
+import prettycron from "prettycron";
+/* eslint-disable no-unused-vars */
+import React from "react";
+/* eslint-enable no-unused-vars */
+import { routerShape } from "react-router";
+import mixin from "reactjs-mixin";
+
+import JobFormModalContainer from "#PLUGINS/jobs/src/js/JobFormModalContainer";
+import TaskStates from "#PLUGINS/services/src/js/constants/TaskStates";
+
+import Icon from "#SRC/js/components/Icon";
+import Page from "#SRC/js/components/Page";
+import TimeAgo from "#SRC/js/components/TimeAgo";
+import TabsMixin from "#SRC/js/mixins/TabsMixin";
+import StringUtil from "#SRC/js/utils/StringUtil";
+import Job from "#SRC/js/structs/Job";
+import JobConfiguration from "./JobConfiguration";
+import { DIALOGS } from "./JobDetailPageContainer";
+import JobRunHistoryTable from "./JobRunHistoryTable";
+
+class JobDetailPage extends mixin(TabsMixin) {
+  constructor() {
+    super(...arguments);
+
+    this.tabs_tabs = {
+      runHistory: "Run History",
+      configuration: "Configuration"
+    };
+
+    this.state = {
+      currentTab: Object.keys(this.tabs_tabs).shift()
+    };
+  }
+
+  getNavigationTabs() {
+    return <ul className="menu-tabbed">{this.tabs_getUnroutedTabs()}</ul>;
+  }
+
+  getDestroyConfirmDialog() {
+    return null;
+  }
+
+  getPrettySchedule(job) {
+    const schedules = job.schedules;
+    if (schedules == null || schedules.length === 0) {
+      return null;
+    }
+
+    const schedule = schedules[0];
+    if (schedule.enabled) {
+      return prettycron.toString(schedule.cron);
+    }
```

Comment:
final return missing? 🤔 (if it is impossible to get here, that is no obvious)

Reply:
Yeah, final return seems to be missing, I only moved that code over, but I can refactor it to use an early return instead 👍